src/VAC/VectorAnimationComplex/InfillPattern.cpp

```diff
@@ -0,0 +1,473 @@
+                inset_.containsPoint(potentialTopPoint, Qt::OddEvenFill) &&
+                inset_.containsPoint(potentialBottomPoint, Qt::OddEvenFill)) {
+            if (up) {
+                infillPoints << potentialBottomPoint;
+                infillPoints << potentialTopPoint;
+            } else {
+                infillPoints << potentialTopPoint;
+                infillPoints << potentialBottomPoint;
+            }
+        }
+        up = !up;
+    }
+
+    data_ << QVector<QVector<QPointF>>{infillPoints};
+}
+
+void InfillPattern::rectiLinearHorizontalInfill()
+{
+    QVector<QPointF> infillPoints{};
+
+    const auto outerBoundingBox = inputPolygon_.boundingRect();
+    const auto startBoundary = outerBoundingBox.top();
+    const auto endBoundary = outerBoundingBox.bottom();
+    const auto left = outerBoundingBox.left();
+    const auto right = outerBoundingBox.right();
+    auto directionRight = true;
+
+    for (auto y = startBoundary; y <= endBoundary; y += spacing_) {
+        auto potentialLeftPoint = QPointF{left, y};
+        while(!inset_.containsPoint(potentialLeftPoint, Qt::OddEvenFill) && potentialLeftPoint.x() <= right) {
+            potentialLeftPoint += QPointF{1, 0};
+        }
+        auto potentialRightPoint = QPointF{right, y};
+        while(!inset_.containsPoint(potentialRightPoint, Qt::OddEvenFill) && potentialRightPoint.x() >= left) {
+            potentialRightPoint -= QPointF{1, 0};
+        }
+
+        if (potentialRightPoint.x() - potentialLeftPoint.x() > spacing_ &&
+                inset_.containsPoint(potentialLeftPoint, Qt::OddEvenFill) &&
+                inset_.containsPoint(potentialRightPoint, Qt::OddEvenFill)) {
+            if (directionRight) {
+                infillPoints << potentialRightPoint;
+                infillPoints << potentialLeftPoint;
+            } else {
+                infillPoints << potentialLeftPoint;
+                infillPoints << potentialRightPoint;
+            }
+        }
+        directionRight = !directionRight;
+    }
+
+    data_ << QVector<QVector<QPointF>>{infillPoints};
+}
+
+inline QPointF rotateAndNormalizePoint(double angle, const QPointF& vector)
+{
+    constexpr auto r = 1;
+    auto phi = atan2(vector.y(), vector.x());
+    phi += angle;
+    return QPointF{r * cos(phi), r * sin(phi)};
+}
+
+QPointF lineLineIntersection(QPointF A, QPointF B, QPointF C, QPointF D)
+{
+    // Line AB represented as a1x + b1y = c1
+    double a1 = B.y() - A.y();
+    double b1 = A.x() - B.x();
+    double c1 = a1*(A.x()) + b1*(A.y());
+
+    // Line CD represented as a2x + b2y = c2
+    double a2 = D.y() - C.y();
+    double b2 = C.x() - D.x();
+    double c2 = a2*(C.x())+ b2*(C.y());
+
+    double determinant = a1*b2 - a2*b1;
+
+    if (determinant == 0)
+    {
+        // The lines are parallel. This is simplified
+        // by returning a pair of FLT_MAX
+        return QPointF(FLT_MAX, FLT_MAX);
```

Comment:
you could use quiet NaN as well

Reply:
I changed to optional. Makes more sense I think. Added a comment as well about it being concave.

Reply:
that's a much better solution! 👍👍